### ISSUE TITLE
SAM API users - Looking for your feedback 

### ISSUE THREAD

**Issue by pammiller0** [GSA/sam_api#61]
All including @kaitlin, @yacu01 - we would like feedback from you as regular users of the SAM API. On January 16th we did a SAM release that introduced new fields (DFARS fields) into the API. Have any of you had any issues using the API since then? As you may have seen in issue #59 it broke @ChrisWagner214 's code however we haven't heard of anyone else having issues. We would like to get your feedback to better understand how you are setting up code and implementing our API. Thanks!

**Comment by ChrisWagner214** [comment]
Are you able to provide any metrics on API use by the API keys that are issued?  This may help to reveal how the API is being used (number of users & number of requests).  Just a thought.

**Comment by pammiller0** [comment]
Yes it is in the plans to provide usage metrics. It's not there yet but it is in our plans to get some useful metrics out there so you guys can see user count and number of requests. Please post any more stats you would like to see. Thanks!

**Comment by kaitlin** [comment]
Do you have a [changelog](http://en.wikipedia.org/wiki/Changelog) of what fields changed? I haven't noticed our scripts breaking but it would be nice to have a record (and potentially a heads up before it actually happens) of what changed. My biggest problem with the SAM API right now is that the hubzone and 8A designators are not accurate.

**Comment by ChrisWagner214** [comment]
@pammiller0  Any idea when will documentation be available for the current release of the API?  What is up on the web at http://gsa.github.io/sam_api/sam looks to be out of date and incomplete.  As a developer, it is very hard to consume this API with the documentation as it is.

Also, the rate limit and volume limit on the API further add to the difficulty of consuming this API.  

Thanks.

**Comment by cloupid** [comment]
I agree.  The docs need a bit of help.  

Some issues:

- v2 endpoint is no where to be found (I just found it through another issue here on github).
- Fields are incorrect (The Country/Country code issue).
- Docs show camelCase for basically everything but the Address object is not (except stateorProvince but according to the docs it should be stateOrProvince)
- If a registration is expired the entire name appears to be in the firstName field for POCs.
- Better explanation of status codes.  For example some previous registrations in the download files will return 403 through the API.  Some registrations will say Expired and some are 404. 

Wish list:

-Search by entity name

**Comment by pammiller0** [comment]
@kaitlin we are going to be incorporating a change log and API roadmap into this site shortly, hopefully within the next couple of weeks. @cloupid  https://api.data.gov/sam/v2/registrations/ is the link to v2 of the SAM API. We will be posting the full v1 and v2 documentation within a week or so. Also, we are checking into the issue you pointed out with expired registrations. I will get back on that one along with posting more information on status messages.

**Comment by baskerGanesan** [comment]
Please treat this feedback as high priority.

when I use getData api to search public data using duns number and if there is no record found/error found for the particular duns number then the api returns 500 internal server error. 
https://api.data.gov/sam/v1/registrations/dunsnumberandapikey
please change the  error status type or exception type as different then internal server error

We struggled alot to understand this and go though this.

**Comment by aromero6** [comment]
@baskerGanesan

After speaking to our support team, this is something they are aware of and
working on resolving quickly.  We are experience an issue with our ESB
servers that affects session handling.  We are working on a fix.  This is
not a user error.

We apologize for this and will keep you updated as progress is made.

Thanks,
Tony


Antonio Romero
US Federal Government
General Services Administration OCIO
Office of Enterprise Infrastructure
O: 312.353.3116
C: 312.497.5592
